samples/gen-go-db/server/db/dynamodb/teachersharingrule.go

```diff
@@ -195,6 +195,68 @@ func (t TeacherSharingRuleTable) getTeacherSharingRulesByTeacherAndSchoolApp(ctx
 	return decodeTeacherSharingRules(queryOutput.Items)
 }
 
+func (t TeacherSharingRuleTable) getTeacherSharingRulesByTeacherAndSchoolAppPage(ctx context.Context, input db.GetTeacherSharingRulesByTeacherAndSchoolAppPageInput, fn func(m *models.TeacherSharingRule, lastTeacherSharingRule bool) bool) error {
+	queryInput := &dynamodb.QueryInput{
+		TableName: aws.String(t.name()),
+		ExpressionAttributeNames: map[string]*string{
+			"#TEACHER":    aws.String("teacher"),
+			"#SCHOOL_APP": aws.String("school_app"),
+		},
+		ExpressionAttributeValues: map[string]*dynamodb.AttributeValue{
+			":teacher": &dynamodb.AttributeValue{
+				S: aws.String(input.StartingAfter.Teacher),
+			},
+			":schoolApp": &dynamodb.AttributeValue{
+				S: aws.String(fmt.Sprintf("%s_%s", input.StartingAfter.School, input.StartingAfter.App)),
+			},
+		},
+		ScanIndexForward: aws.Bool(!input.Descending),
+		ConsistentRead:   aws.Bool(!input.DisableConsistentRead),
+		Limit:            input.Limit,
+		ExclusiveStartKey: map[string]*dynamodb.AttributeValue{
+			"school_app": &dynamodb.AttributeValue{
+				S: aws.String(fmt.Sprintf("%s_%s", input.StartingAfter.School, input.StartingAfter.App)),
+			},
+			"teacher": &dynamodb.AttributeValue{
+				S: aws.String(input.StartingAfter.Teacher),
+			},
+		},
+	}
+	if input.Descending {
+		queryInput.KeyConditionExpression = aws.String("#TEACHER = :teacher AND #SCHOOL_APP <= :schoolApp")
+	} else {
+		queryInput.KeyConditionExpression = aws.String("#TEACHER = :teacher AND #SCHOOL_APP >= :schoolApp")
+	}
+
+	queryOutput, err := t.DynamoDBAPI.QueryWithContext(ctx, queryInput)
+	if err != nil {
+		return err
+	}
+	if len(queryOutput.Items) == 0 {
+		fn(nil, false)
```

Comment:
curious what's the reasoning here? I'd be surprised receiving a `nil` model in the callback.

Reply:
I do not remember. It doesn't make sense.